vendor/github.com/containers/image/v5/docker/docker_transport.go

```diff
@@ -51,11 +51,22 @@ func ParseReference(refString string) (types.ImageReference, error) {
 	if !strings.HasPrefix(refString, "//") {
 		return nil, fmt.Errorf("docker: image reference %s does not start with //", refString)
 	}
+	// Check if ref has DigestOnlyTag suffixed to it
+	digestOnly := false
+	if strings.HasSuffix(refString, reference.DigestOnlyTag) {
+		digestOnly = true
+		refString = strings.TrimSuffix(refString, reference.DigestOnlyTag)
```

Comment:
`HasSuffix` would match `example.com/reponame-ends-with-digest-only`, as well as `example.com/repo:tag-ends-with-digest-only`. That’s far too broad.

Reply:
Right, do you have any recommendation on a specific suffix that we can use that would most probably not be used by users? I feel like any form of string could be a risk, but I couldn't come up with anything better.

Reply:
(For the design as currently implemented, with a reserved tag value, this should be a `reference.Digested` or `reference.Canonical` type check, relying on the existing parser, instead of doing any string matching. But I think a completely new syntax would be better.)

Maybe something like `@@digest-unspecified`?

- `@` is forbidden in repo names, and separates repo@digest, so the repo vs. marker split is unambiguous
- The text after `@` has, so far, been considered a `digest.Digest` string, and those values should match `reference.DigestRegexp`; using another `@` ensures it can’t match, which makes it unambiguous.
- (Using the `@` marker is… somewhat compatible :) with incorrect parsers who look for `@` to see a digested reference, without parsing it correctly and noticing that it is invalid. Arguably that’s actually a disadvantage.)

Alternatively, something like `:@digest-unspecified`; again, `:` serves as an already-defined separator between repo vs. tag; and tags must match `reference.TagRegexp`.

Reply:
Got it, fixed to use `@@unknown-digest@@` as the suffix for the digest only use case and it is what we check for in `ParseReference`. I also removed the functionality of adding a tag with this value to the image name.